lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2470,6 +2470,15 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
         }
       }
     }
+
+    // @_staticExclusiveOnly types cannot be put into 'var's, only 'let'.
+    if (auto SD = VD->getInterfaceType()->getStructOrBoundGenericStruct()) {
```

Comment:
This should be `dyn_cast<StructDecl>(VD)` instead

Reply:
But this is checking that the var decl is a struct decl which is incorrect no? I want to get the var's type and see if that type has this attribute.